app/src/backend/CSVDownloadService.ts

```diff
@@ -98,95 +137,112 @@ export default class CSVDownloadService {
-          gpc_reference_number: inventoryValue.gpcReferenceNumber,
-          subsector_name: inventoryValue.subSector.subsectorName,
-          notation_key: inventoryValue.unavailableReason?.split("-")[1],
-          activityValues: activityValues.map((activityValue) => {
-            let activityTitleKey = activityValue.metadata?.activityTitle;
-            let dataQuality = activityValue.metadata?.dataQuality;
-            let dataSource = activityValue.activityData?.["data-source"];
-            let activityAmount = activityValue.activityData?.[activityTitleKey];
-            let activityUnit = t(
-              activityValue.activityData?.[`${activityTitleKey}-unit`],
-            );
-            let emission_co2 = null;
-            let emission_ch4 = null;
-            let emission_n2o = null;
-
-            if (activityValue.gasValues) {
-              let co2_gas = activityValue.gasValues.find(
-                (g) => g.gas === "CO2",
-              );
-              let ch4_gas = activityValue.gasValues.find(
-                (g) => g.gas === "CH4",
-              );
-              let n2o_gas = activityValue.gasValues.find(
-                (g) => g.gas === "N2O",
-              );
-
-              emission_co2 = co2_gas?.emissionsFactor.emissionsPerActivity;
-              emission_ch4 = ch4_gas?.emissionsFactor.emissionsPerActivity;
-              emission_n2o = n2o_gas?.emissionsFactor.emissionsPerActivity;
-            }
-            // if there is an existing emission factor value
-            let usesEf =
-              inventoryValue.gpcReferenceNumber?.split(".")?.includes("I") ||
-              inventoryValue.gpcReferenceNumber?.split(".").includes("II");
-            let efUnit = null;
-
-            if (usesEf) {
-              let scope = inventoryValue.gpcReferenceNumber?.split(".")[2];
-              efUnit = scope === "1" ? "kg/m3" : "kg/TJ";
-            }
-
-            return {
-              emission_factor_unit: efUnit,
-              emission_co2,
-              emission_ch4,
-              emission_n2o,
-              activity_amount: activityAmount,
-              activity_unit: activityUnit,
-              data_source_id: "",
-              data_source_name: dataSource,
-              total_co2e: toDecimal(activityValue.co2eq as bigint)
-                ?.div(new Decimal("1e3"))
-                .toNumber(),
-            };
-          }),
-        };
-      }
+      const activityTypeKey =
+        activityTypeKeyMapping[inventoryValue.gpcReferenceNumber as string]?.[
+          inventoryValue.inputMethodology as string
+        ];
+
+      dataDictionary[gpcRefNo as string] = {
+        // InventoryValue fields
+        inventory_reference: inventoryValue.subCategoryId,
+        gpc_reference_number: inventoryValue.gpcReferenceNumber,
+        subsector_name: inventoryValue.subSector.subsectorName,
+        notation_key: inventoryValue.unavailableReason?.split("-")[1],
```

Comment:
```suggestion
        notation_key: inventoryValue.unavailableReason?.split("-")?.[1],
```
To prevent a null error in case unavailableReason is null.